src/Cli/ConfigGenerator.cs

```diff
@@ -1091,6 +1082,63 @@ public static bool TryStartEngineWithOptions(StartOptions options, FileSystemRun
             return Azure.DataApiBuilder.Service.Program.StartEngine(args.ToArray());
         }
 
+        /// <summary>
+        /// Runs all the validations on the config file and returns true if the config is valid.
+        /// </summary>
+        public static bool IsConfigValid(ValidateOptions options, FileSystemRuntimeConfigLoader loader, IFileSystem fileSystem)
```

Comment:
similar comment as above, the `FileSystemRuntimConfigLoader` has a `IFileSystem` is there a reason we aren't just grabbing that filesystem? convenience?

Reply:
yes, i had this question as well, after checking when the fileSystem was introduced, it was like this and it was passing both loader and fileSystem separately. I believe the fileSystem is intended to be used only within the class and not modified externally, so it was better to keep it private and not create a getter or something.